ConverterViewController.swift

```diff
@@ -0,0 +1,174 @@
+    @IBOutlet weak var lblJPY: UILabel!
+    
+    @IBOutlet weak var dollarNumberTextField: UITextField!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        //sets format number style to currency
+        _currencyFormatter.numberStyle = NSNumberFormatterStyle.CurrencyStyle
+        
+        //set delegate for dollarnumberUITextField
+        self.dollarNumberTextField.delegate = self
+        
+        //get current dollar price from fixer.io
+        getCurrencyValues(self.updateValues)
+        
+        // Gesture that recognices tap to hide keyboard
+        let tapGesture = UITapGestureRecognizer(target: self, action:Selector("endEditing:"))
+        tapGesture.cancelsTouchesInView = false
+        self.view.addGestureRecognizer(tapGesture);
+    }
+    
+    // fetches exchange values for dollar using a get request and receiving a callback
+    func getCurrencyValues(completion: (() -> Void)!) {
+        
+        // Create NSURL Ibject
+        let pricesUrl = NSURL(string: EXCHANGESURL);
+        
+        // Creaste URL Request
+        let pricesRequest = NSMutableURLRequest(URL:pricesUrl!);
+        
+        // Set request HTTP method to GET. It could be POST as well
+        pricesRequest.HTTPMethod = REQUESTMETHOD
+        
+        // Excute HTTP Request
+        let task = NSURLSession.sharedSession().dataTaskWithRequest(pricesRequest) {
+            data, response, error in
+            
+            // Check for error
+            if error != nil
+            {
+                print("error=\(error)")
+                return
+            }
+            
+            // Print out response string
+            let responseString = NSString(data: data!, encoding: NSUTF8StringEncoding)
+            print("responseString = \(responseString)")
+            
+            
+            // Convert server json response to NSDictionary
+            do {
+                if let convertedJsonIntoDict = try NSJSONSerialization.JSONObjectWithData(data!, options: []) as? NSDictionary {
+                    
+                    self._requestedWasOk = true
+                    
+                    self._exchangesDictionary = convertedJsonIntoDict["rates"] as? [String : Double]
+                    
+                    completion() //callback
+                    
+                }
+            } catch let error as NSError {
+                print(error.localizedDescription)
+                
+                self._requestedWasOk = false
+            }
+            
+        }
+        
+        task.resume()
+    }
+    
+    //displays equivalent values in different format currencies.
+    func updateValues() {
+     
+      dispatch_async(dispatch_get_main_queue(), {
+        
+            //Calculates and shows exchange value for UK Pounds
+            if var priceGBP = self._exchangesDictionary![CURRENCIES["UK"]!] {
+                priceGBP = priceGBP * self._dollarAmount!;
```

Comment:
NEVER use "!" to unwrap conditionals.   This will cause the application to crash if there is a nil value.   Use "let" and "guard".